main/exercise/calculated_answer.class.php

```diff
@@ -25,119 +25,202 @@ public function __construct()
+     * for blanks : [[name, min, max, digit]*]
+     * for formulas : [name, content, tolerance number, tolerance type percent|digit, number of digit max, score]*]
+     *
+     * @param Answer $answerObject
+     * @return array
+     */
+    public static function parseCalculatedAnswer(Answer $answerObject)
+    {
+        $blankList = [];
+        $formulaList = [];
+
+        // get the last pattern found
+        list($wordingAnswer, $encodedAnswer) = self::getEditorPart($answerObject->selectAnswer(1));
+
+        // split answer by ;
+        // get the list of formulas and blanks
+        $unknownList = api_preg_split("/;/", $encodedAnswer);
+        foreach ($unknownList as $unknownInfo) {
+            if (api_preg_match("/^=/", $unknownInfo)) {
+                $tempFormulaInfos = self::parseCalculatedAnswerFormula($unknownInfo);
+                $formulaList[$tempFormulaInfos['name']] = $tempFormulaInfos;
+            } else if (api_preg_match("/^#/", $unknownInfo)) {
+                $tempBlankInfos = self::parseCalculatedAnswerBlank($unknownInfo);
+                $blankList[$tempBlankInfos['name']] = $tempBlankInfos;
             }
+        }
+        return [$wordingAnswer, $blankList, $formulaList];
+    }
+
+
+    /**
+     * return defaults values for question form
+     * @param Answer $answerObject
+     * @return array
+     */
+    public static function getDefaultsCreateAnswerForm(Answer $answerObject)
+    {
+        $defaults = [];
+        list($answerText, $blankList, $formulaList) = self::parseCalculatedAnswer($answerObject);
+
+        $defaults['answer'] = $answerText;
+
+        foreach ($blankList as $blankInfo) {
+            $defaults["blank_min_".$blankInfo["name"]] = $blankInfo["min"];
+            $defaults["blank_max_".$blankInfo["name"]] = $blankInfo["max"];
+            $defaults["blank_decimals_".$blankInfo["name"]] = $blankInfo["digit"];
+        }
+
+        foreach ($formulaList as $formulaInfo) {
+            $defaults["formula_content_".$formulaInfo["name"]] = $formulaInfo["content"];
+            $defaults["formula_".$formulaInfo["name"]] = $formulaInfo["content"];
+            $defaults["formula_tolerance_".$formulaInfo["name"]] = $formulaInfo["tolerance"];
+            $defaults["formula_tolerancetype_".$formulaInfo["name"]] = $formulaInfo["toleranceType"];
+            $defaults["formula_digit_".$formulaInfo["name"]] = $formulaInfo["digit"];
+            $defaults["formula_score_".$formulaInfo["name"]] = $formulaInfo["score"];
+        }
+
+        return $defaults;
+    }
+
+
+    /**
+     * function which redefines Question::createAnswersForm
+     * @param FormValidator $form
+     */
+    public function createAnswersForm($form)
+    {
+        $defaults = array();
 
-            window.onload = updateBlanks;
+        // javascript
+        echo "<script>ajaxTextPath = '".api_get_path(WEB_AJAX_PATH)."text.ajax.php';</script>";
+        echo api_get_js("chance/chance.min.js");
```

Comment:
You are apparently depending on a "chance.min.js" file that is not present in your PR!